From d447907240076223f05d916a6a9f8f656cafa96a Mon Sep 17 00:00:00 2001
From: Gaku Inami <gaku.inami.xw@bp.renesas.com>
Date: Wed, 25 Jun 2014 16:58:40 +0900
Subject: ARM: shmobile: Remove opps table check for cpufreq

This patch is based on feedback from Viresh Kumar.
Since cpufreq-cpu0 driver has already check opp table, there is no
need to same check in mach-shmobile.

Signed-off-by: Gaku Inami <gaku.inami.xw@bp.renesas.com>
Acked-by: Viresh Kumar <viresh.kumar@linaro.org>
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit e423d12cbcb9576785e891617888f627f1f57bf4)
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
---
 arch/arm/mach-shmobile/cpufreq.c | 16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/arch/arm/mach-shmobile/cpufreq.c b/arch/arm/mach-shmobile/cpufreq.c
index e2c868fc41cf..8a24b2be46ae 100644
--- a/arch/arm/mach-shmobile/cpufreq.c
+++ b/arch/arm/mach-shmobile/cpufreq.c
@@ -8,24 +8,10 @@
  * for more details.
  */
 
-#include <linux/of.h>
-#include <linux/of_device.h>
 #include <linux/platform_device.h>
 
 int __init shmobile_cpufreq_init(void)
 {
-	struct device_node *np;
-
-	np = of_cpu_device_node_get(0);
-	if (np == NULL) {
-		pr_err("failed to find cpu0 node\n");
-		return 0;
-	}
-
-	if (of_get_property(np, "operating-points", NULL))
-		platform_device_register_simple("cpufreq-cpu0", -1, NULL, 0);
-
-	of_node_put(np);
-
+	platform_device_register_simple("cpufreq-cpu0", -1, NULL, 0);
 	return 0;
 }
-- 
2.1.2

